pkg/sdn/plugin/ovscontroller.go

```diff
@@ -104,12 +112,21 @@ func (oc *ovsController) SetupOVS(clusterNetworkCIDR, serviceNetworkCIDR, localS
 	// Table 21: from OpenShift container; NetworkPolicy plugin uses this for connection tracking
 	otx.AddFlow("table=21, priority=0, actions=goto_table:30")
 
+	if oc.useConnTrack {
+		// Table 25: IP from OpenShift container via Service IP; reload tenant-id; filled in by openshift-sdn-ovs
+		// eg, "table=25, priority=100, ip, nw_src=${ipaddr}, actions=load:${tenant_id}->NXM_NX_REG0[], goto_table:30"
+		otx.AddFlow("table=25, priority=0, actions=drop")
+	}
+
 	// Table 30: general routing
 	otx.AddFlow("table=30, priority=300, arp, nw_dst=%s, actions=output:2", localSubnetGateway)
 	otx.AddFlow("table=30, priority=200, arp, nw_dst=%s, actions=goto_table:40", localSubnetCIDR)
 	otx.AddFlow("table=30, priority=100, arp, nw_dst=%s, actions=goto_table:50", clusterNetworkCIDR)
 	otx.AddFlow("table=30, priority=300, ip, nw_dst=%s, actions=output:2", localSubnetGateway)
 	otx.AddFlow("table=30, priority=100, ip, nw_dst=%s, actions=goto_table:60", serviceNetworkCIDR)
+	if oc.useConnTrack {
+		otx.AddFlow("table=30, priority=300, ip, nw_dst=%s, ct_state=+rpl, actions=ct(nat),goto_table:70", localSubnetCIDR)
```

Comment:
This is kind of a hack; we call ct(nat) on *all* pod-reply-to-pod packets; if they're DNATted this will undo that and if they're not it's a no-op. In earlier versions of the patch we were setting ct_state on the outgoing packets and then only un-DNATting the replies with ct_state set. But then we have to worry about our use of ct_state conflicting with other uses, etc. So this avoids that